### PR TITLE
Boosting fix

### PR DESCRIPTION
**Pull request by shahafn**

### CODE REVIEW

File: dockers/relaydc/docker-compose.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ services:
 #### make sure to add instance name to the router command above
 
   gsn1:
-    image: opengsn/jsrelay:2.2.2
+    image: opengsn/jsrelay:2.2.3-matic
```
[the remark this answers is not in the document]
Comment (shahafn) [reply dockers/relaydc/docker-compose.yml:35]: Right

File: packages/common/src/ContractInteractor.ts [hunk 2]
```diff
@@ -485,19 +479,55 @@ export class ContractInteractor {
       // this is to avoid '!' for TypeScript
       options.toBlock = 'latest'
     }
+    if (options.fromBlock == null) {
+      options.fromBlock = 1
```
[the remark this answers is not in the document]
Comment (shahafn) [reply packages/common/src/ContractInteractor.ts:483]: That behavior was already there, I only fixed error checking

File: packages/common/src/ContractInteractor.ts [hunk 3]
```diff
@@ -485,19 +479,55 @@ export class ContractInteractor {
...
+    }
     // save 'getBlockNumber' roundtrip for a known max value (must match check in getLogsPagesForRange)
     if (this.maxPageSize !== Number.MAX_SAFE_INTEGER && options.toBlock === 'latest') {
       options.toBlock = await this.getBlockNumber()
+      if (options.fromBlock > options.toBlock) {
+        options.toBlock = options.fromBlock
```
[the remark this answers is not in the document]
Comment (shahafn) [reply packages/common/src/ContractInteractor.ts:489]: Yes, it was there before, but I'll fix it here

File: packages/dev/test/RelayServerRequestsProfiling.test.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@ import { GSNContractsDeployment } from '@opengsn/common/dist/GSNContractsDeploym
 
 contract('RelayServerRequestsProfiling', function (accounts) {
   const refreshStateTimeoutBlocks = 2
-  const callsPerStateRefresh = 9
+  const callsPerStateRefresh = 10
```
[the remark this answers is not in the document]
Comment (shahafn) [reply packages/dev/test/RelayServerRequestsProfiling.test.ts:13]: It was 9, now there's another call, to getTransactionCount()

File: packages/relay/src/runServer.ts [hunk 5]
```diff
@@ -42,6 +42,27 @@ async function run (): Promise<void> {
       error('missing ethereumNodeUrl')
     }
     web3provider = new Web3.providers.HttpProvider(conf.ethereumNodeUrl)
+    // TEMP: logging provider..
```
[the remark this answers is not in the document]
Comment (shahafn) [reply packages/relay/src/runServer.ts:45]: I'm not sure, it's very useful for me to uncomment it when it's needed for debugging. Maybe I'll add a debug flag that would do it

File: packages/relay/src/runServer.ts [hunk 6]
```diff
@@ -28,6 +28,8 @@ function error (err: string): never {
   process.exit(1)
 }
 
+const DEBUG_PROVIDER = true
```
Comment (drortirosh) [inline packages/relay/src/runServer.ts:31]: don't default it to true..
(maybe add it to config? this way it can be enabled on a real, running relayer.)

Comment (shahafn) [reply packages/relay/src/runServer.ts]: Added to config

**Review by drortirosh — APPROVED** [review]

**Pull request merged**